Modules/_ssl.c

```diff
@@ -4109,7 +4109,7 @@ _ssl_MemoryBIO_read_impl(PySSLMemoryBIO *self, int len)
     int avail, nbytes;
     PyObject *result;
 
-    avail = BIO_ctrl_pending(self->bio);
+    avail = (int)BIO_ctrl_pending(self->bio);
```

Comment:
`BIO_read` takes an `int` so we can only really only read as much as an `int` can hold without calling `BIO_read` in a loop. I replaced this with a `Py_MIN` which will make us read up to `INT_MAX`.